src/mgmt/srv.c

```diff
@@ -514,11 +534,28 @@ process_drpc_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	 */
 	switch (drpc_req->method) {
 	case DRPC_METHOD_MGMT_KILL_RANK:
-		process_killrank_request(drpc_req, daos_resp);
+		/* Unpack the daos request from the drpc call body */
```

Comment:
Per our discussion earlier, I'm not crazy about putting a lot of logic inline in the switch block. IMO each case could be a helper method. I do like the way you broke out the logic of the actual work though.